Source/PeleLMeX_Diffusion.cpp

```diff
@@ -206,6 +206,171 @@ PeleLM::computeDifferentialDiffusionTerms(
+    int doZeroVisc = 1;
+    int addTurbContribution = 0;
+    Array<MultiFab, AMREX_SPACEDIM> beta_ec = getDiffusivity(
+      lev, 0, NUM_SPECIES, doZeroVisc, bcRecSpec, ldata_beta_cc,
+      addTurbContribution);
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+    for (MFIter mfi(ldata_beta_cc, TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+      for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        const auto bc_lo = m_phys_bc.lo(idim);
+        const auto bc_hi = m_phys_bc.hi(idim);
+        const Box& edomain = amrex::surroundingNodes(domain, idim);
+        const Box& ebx = mfi.nodaltilebox(idim);
+        auto const& flux_soret = soretfluxes[lev][idim]->const_array(mfi);
+        auto const& rhoD_ec = beta_ec[idim].const_array(mfi);
+        auto const& flux_wbar = (m_use_wbar != 0 && !need_explicit_fluxes)
+                                  ? a_wbarfluxes[lev][idim]->const_array(mfi)
+                                  : rhoD_ec;
+        auto const& boundary_ar = a_spec_boundary[lev]->array(mfi);
+        amrex::ParallelFor(
+          ebx,
+          [flux_wbar, flux_soret, rhoD_ec, boundary_ar, idim, edomain, bc_lo,
+           bc_hi, use_wbar = m_use_wbar,
+           need_explicit_fluxes] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+            int idx[3] = {i, j, k};
+            bool on_lo = (bc_lo == BoundaryCondition::BCNoSlipWallIsotherm ||
+                          bc_lo == BoundaryCondition::BCSlipWallIsotherm) &&
+                         (idx[idim] <= edomain.smallEnd(idim));
+            bool on_hi = (bc_hi == BoundaryCondition::BCNoSlipWallIsotherm ||
+                          bc_hi == BoundaryCondition::BCSlipWallIsotherm) &&
+                         (idx[idim] >= edomain.bigEnd(idim));
+            if (on_lo || on_hi) {
+              if (on_lo) { // need to move -1 for lo boundary
+                idx[idim] -= 1;
+              }
+              for (int n = 0; n < NUM_SPECIES; n++) {
+                boundary_ar(idx[0], idx[1], idx[2], n) = flux_soret(i, j, k, n);
+                // add lagged wbar flux
+                if (use_wbar != 0 && !need_explicit_fluxes) {
+                  boundary_ar(idx[0], idx[1], idx[2], n) +=
+                    flux_wbar(i, j, k, n);
+                }
+                boundary_ar(idx[0], idx[1], idx[2], n) /= rhoD_ec(i, j, k, n);
+              }
+            }
+          });
+      }
+    }
+  }
+  if (need_explicit_fluxes && m_use_wbar != 0) { // need to iterate to get the
+                                                 // proper wbar flux
+    Vector<Array<MultiFab, AMREX_SPACEDIM>> wbarfluxes(finest_level + 1);
+    for (int lev = 0; lev <= finest_level; lev++) {
+      for (int idim = 0; idim < AMREX_SPACEDIM; idim++) {
+        wbarfluxes[lev][idim].define(
+          grids[lev], dmap[lev], NUM_SPECIES, 1, MFInfo(), Factory(lev));
+        wbarfluxes[lev][idim].setVal(0.0);
+      }
+    }
+    int iter_max = 10;
```

Comment:
Currently 10 iterations hardcoded - we could do some sort of residual check instead

Reply:
10 seems a bit excessive, no?

Reply:
Probably - based on the test case it converged in 2 or 3 iterations